voicevox_engine/morphing.py

```diff
@@ -43,6 +47,53 @@ def create_morphing_parameter(
     )
 
 
+def is_synthesis_morphing_permitted(
+    engine: SynthesisEngineBase, base_speaker: int, target_speaker: int
+) -> Optional[bool]:
+    """
+    指定されたspeakerがモーフィング可能かどうか返す
+    speakerが見つからない場合はNoneを返す
+    """
+
+    speakers = json.loads(engine.speakers)
+    base_speaker_info, target_speaker_info = None, None
+    for speaker in speakers:
+        style_id_arr = tuple(map(lambda style: style["id"], speaker["styles"]))
```

Comment:
一瞬読めませんでした。。。 😇 
なぜglobalにしかmapがないのか、なぜmap(object, func)ではないのか。。。

実は内包表記のが簡潔だったりします。
```suggestion
        style_id_arr = tuple([stayle["id"] for style in speaker["styles"]])
```

Reply:
> ```python
> style_id_arr = tuple(style["id"] for style in speaker["styles"])
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> に修正します

あ、記憶違いかもなんですが、それだとiteratorが１つ含まれたtupleになるはず…？

Reply:
```
In [1]: tuple(a for a in "abc")
Out[1]: ('a', 'b', 'c')
```
大丈夫そう

Reply:
あれ！記憶違いでした、すみません！！